scheduler/util.go

```diff
@@ -376,11 +377,18 @@ func taintedNodes(state State, allocs []*structs.Allocation) (map[string]*struct
 	return out, nil
 }
 
-// shuffleNodes randomizes the slice order with the Fisher-Yates algorithm
-func shuffleNodes(nodes []*structs.Node) {
+// shuffleNodes randomizes the slice order with the Fisher-Yates
+// algorithm. We seed the random source with the eval ID (which is
+// random) to aid in postmortem debuggging of specific evaluations and
+// state snapshots.
```

Comment:
@schmichael noted (elsewhere):
> Will this cause the node iteration order to be the same every time we retry processing the eval? Or would we maintain the RNG state between retries so that each retry would shuffle based on the "next" random value?

It'd be the same seed on each retry. There's not an easy way to thread the retry state down, but given that the plan has to make progress to be retried (I think?), we could use the counts on the current iteration of the plan to "wiggle" the seed we get from the eval ID deterministically too.